### PR TITLE
Feature/20 implement scene

### LINKED ISSUE #20: Create & Implement Scene struct
Should have relevant methods such as update(), which contains the physics loop

### PR DESCRIPTION
**Pull request by isakvlarsson**
Closes #20, you fricks

### CODE REVIEW

File: pphysx3d/src/scene/mod.rs [hunk 1]
```diff
@@ -1 +1,91 @@
...
+    /// Add a GameObject to the scene
+    pub fn add(&mut self, object: GameObject) {
+        self.objects.push(object);
+    }
+
+    /// Remove a GameObject by index from the scene
+    pub fn remove(&mut self, index: usize) {
+        self.objects.remove(index);
+    }
+
```
Comment (M-Blomqvist) [inline pphysx3d/src/scene/mod.rs:26]:
```suggestion

    pub fn objects(&self) -> &[GameObject] {
        &self.objects
    }

```

File: pphysx3d/src/scene/tests.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    };
+    let object_2 = GameObject {
+        shape: sphere_2,
+        position: transform_2,
+        velocity: Vector3::new(0f32, 0f32, 0f32),
+        mass: 0f32,
+    };
+    let objects = vec![object_1, object_2];
+    let collisions = broad_phase(&objects);
+    //println!("{:?}", collisions.len());
```
Comment (M-Blomqvist) [inline pphysx3d/src/scene/tests.rs:68]:
```suggestion
```

File: pphysx3d/src/scene/tests.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    assert_eq!(collisions.len(), 1);
+    assert_eq!(0, collisions[0].0);
+    assert_eq!(1, collisions[0].1);
+    //print!("{:?} \n {:?}\n", objects[0].compute_aabb(), objects[1].compute_aabb())
```
Comment (M-Blomqvist) [inline pphysx3d/src/scene/tests.rs:72]:
```suggestion
```

**Review by M-Blomqvist — CHANGES_REQUESTED** [review]
Pretty good, as a start. In a couple of years you'll be a great programmer

**Review by M-Blomqvist — APPROVED** [review]

**Review by coffecup25 — APPROVED** [review]

**Pull request merged**